sklearn/ensemble/gradient_boosting.py

```diff
@@ -1180,7 +1180,7 @@ def feature_importances_(self):
     def _validate_y(self, y):
         self.n_classes_ = 1
         if y.dtype.kind == 'O':
-            y = y.astype(np.float64)
+            y = astype(y, np.float64, copy=False)
```

Comment:
When `y.dtype.kind == 'O'`, it means that y is an array of Python `object`s. There no need to set `copy=False` as there is no way to avoid the malloc / dtype conversion in that case. Please revert this change.